design/Implemented/supporting-volumeattributes-resource-policy.md 

```diff
@@ -0,0 +1,72 @@
+# Adding Support For VolumeAttributes in Resource Policy
+ 
+## Abstract
+Currently [Velero Resource policies](https://velero.io/docs/main/resource-filtering/#creating-resource-policies) are only supporting "Driver" to be filtered for [CSI volume conditions](https://github.com/vmware-tanzu/velero/blob/8e23752a6ea83f101bd94a69dcf17f519a805388/internal/resourcepolicies/volume_resources_validator.go#L28)
+
+If user want to skip certain CSI volumes based on other volume attributes like protocol or SKU, etc, they can't do it with the current Velero resource policies. It would be convenient if Velero resource policies could be extended to filter on volume attributes along with existing driver filter in the resource policies `conditions` to handle the backup of volumes just by `some specific volumes attributes conditions`.
+ 
+## Background
+As of Today, Velero resource policy already provides us the way to filter volumes based on the `driver` name. But it's not enough to handle the volumes based on other volume attributes like protocol, SKU, etc.
+
+## Example:
+  - Provision Azure NFS: Define the Storage class with `protocol: nfs` under storage class parameters to provision [CSI NFS Azure File Shares](https://learn.microsoft.com/en-us/azure/aks/azure-files-csi#nfs-file-shares).
+  - User wants to back up AFS (Azure file shares) but only want to backup `SMB` type of file share volumes and not `NFS` file share volumes.
+ 
+## Goals
+- Introducing support for `VolumeAttributes` filter along with `driver` filter in CSI volume conditions to handle volumes.
+ 
+## Non-Goals
+- We are only bringing additional support to only handle volumes for backup and do not support restore.
```

Comment:
So user will manually on a needed basis create volume from backup outside of `velero restore`?

Reply:
Can you write a little scenario/example how user would make use of this backup that is not directly restorable in velero?

Reply:
I meant that this was an addition on top of existing Resource Policy design which is supported _only_ during Backup and similar resource policy design is not supported during restore.

Reframing it to: "We are only bringing additional support in the resource policy to only handle volumes during backup."

Reply:
Could you clarify that the volume can be restore, just the filter is only supported on backup similarly to other resource filters?